mods/default/player.lua

```diff
@@ -71,8 +71,15 @@ end
 
 function default.player_set_textures(player, textures)
 	local name = player:get_player_name()
+	local model = models[player_model[name]]
+	local model_textures
+	if model then
+		model_textures = model.textures
+	else
+		model_textures = nil
+	end
 	player_textures[name] = textures
```

Comment:
Why not update this table with the new texture as well? -> `textures or model_textures`